server/controllers.js

```diff
@@ -1,23 +1,12 @@
 const models = require('./models.js');
 
 module.exports = {
-  getProducts: (req, res) => {
-    models.getProducts(req)
-      .then(({ data }) => {
-        res.status(200);
-        res.end(JSON.stringify(data));
-      })
-      .catach((err) => {
-        console.log(err);
-        res.sendStatus(404);
-      });
-  },
-
   getProduct: (req, res) => {
-    models.getProduct(req)
-      .then(({ data }) => {
-        res.status(200);
-        res.end(JSON.stringify(data));
+    models.getProduct(req.params.product_id)
```

Comment:
interesting, I used the query object attached to the request (req.query) vs the parameters (req.params) - I wonder what the difference is? A basic search tells me that query allows you to access more specific parts of the request url (e.g number of responses/page and which page to start on) but seeing as you only need one product at a time, this might not matter for you!

Reply:
I used the params object because the product_id is passed in the URL of the API request. It has to do with the nature of how I set up the routes and MVC pathways, but query is also a good option!